hudi-common/src/main/java/org/apache/hudi/common/config/TypedProperties.java

```diff
@@ -44,27 +43,22 @@ public TypedProperties(Properties defaults) {
   }
 
   private void checkKey(String property) {
-    if (!keyExists(property)) {
+    if (!containsKey(property)) {
       throw new IllegalArgumentException("Property " + property + " not found");
```

Comment:
The old code requires that the key should be a string, what about the new code then ?

Reply:
You mean we can support key with different type? After this, TypedProperties also can only accept keys which is strings, because all public methods can only accept key as String

Reply:
Can we explain why we get performance promotion here ? The `TypedProperties` should not be a hotspot method right ?

Reply:
The `TypedProperties` could be a hotspot method if we use `CustomKeyGenerator` to do the bulk insert, because it will create `SimpleKeyGenerator` and `ComplexKeyGenerator` when `getPartitionPath` and `getRecordKey`(these two methods will called for every rows in the dataset), creating these generators will call its' parent constructor method,

```java
protected BaseKeyGenerator(TypedProperties config) {
    super(config);
    this.encodePartitionPath = config.getBoolean(KeyGeneratorOptions.URL_ENCODE_PARTITIONING.key(),
        Boolean.parseBoolean(KeyGeneratorOptions.URL_ENCODE_PARTITIONING.defaultValue()));
    this.hiveStylePartitioning = config.getBoolean(KeyGeneratorOptions.HIVE_STYLE_PARTITIONING_ENABLE.key(),
        Boolean.parseBoolean(KeyGeneratorOptions.HIVE_STYLE_PARTITIONING_ENABLE.defaultValue()));
    this.consistentLogicalTimestampEnabled = config.getBoolean(KeyGeneratorOptions.KEYGENERATOR_CONSISTENT_LOGICAL_TIMESTAMP_ENABLED.key(),
        Boolean.parseBoolean(KeyGeneratorOptions.KEYGENERATOR_CONSISTENT_LOGICAL_TIMESTAMP_ENABLED.defaultValue()));
  }
```
As `TypedProperties.getBolean` will call `keyExists`, which copy all items to do the check.

Reply:
Thanks, make sense.